### DISCUSSION TITLE
Nuget restore error with sdk:6.0-alpine-amd64

### DISCUSSION THREAD

**Discussion by mcraiha** [dotnet/dotnet-docker#3839] [General]
### Describe the Bug

I have run one of my dotnet project's build process with **mcr.microsoft.com/dotnet/sdk:6.0-alpine-amd64** and at least four months ago it worked just fine. Today I tried it again, and I am getting following error
```
Microsoft (R) Build Engine version 17.2.0+41abc5629 for .NET
Copyright (C) Microsoft Corporation. All rights reserved.

  Determining projects to restore...
/source/WhisperDragon_CLI.csproj : error NU1301: Unable to load the service index for source https://api.nuget.org/v3/index.json.
  Failed to restore /source/WhisperDragon_CLI.csproj (in 5.77 sec).
The command '/bin/sh -c dotnet publish -c release -r linux-musl-x64 --self-contained -o /app -p:PublishSingleFile=true -p:EnableCompressionInSingleFile=true' returned a non-zero code: 1
```

### Steps to Reproduce

1. Clone repo https://github.com/mcraiha/WhisperDragonCLI
2. Run `docker build .`

### Other Information

I also tried with **mcr.microsoft.com/dotnet/sdk:6.0-alpine3.15-amd64** and got same error. I can download the file on host just fine with `wget https://api.nuget.org/v3/index.json`

### Output of `docker version`
```
Client: Docker Engine - Community
 Version:           20.10.17
 API version:       1.41
 Go version:        go1.17.11
 Git commit:        100c701
 Built:             Mon Jun  6 23:02:57 2022
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.17
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.11
  Git commit:       a89b842
  Built:            Mon Jun  6 23:01:03 2022
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.6.6
  GitCommit:        10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc:
  Version:          1.1.2
  GitCommit:        v1.1.2-0-ga916309
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```
### Output of `docker info`

```
Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.8.2-docker)
  scan: Docker Scan (Docker Inc., v0.17.0)

Server:
 Containers: 20
  Running: 0
  Paused: 0
  Stopped: 20
 Images: 8
 Server Version: 20.10.17
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: runc io.containerd.runc.v2 io.containerd.runtime.v1.linux
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 10c12954828e7c7c9b6e0ea9b0c02b01407d3ae1
 runc version: v1.1.2-0-ga916309
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
 Kernel Version: 5.4.72-microsoft-standard-WSL2
 Operating System: Ubuntu 20.04.4 LTS
 OSType: linux
 Architecture: x86_64
 CPUs: 16
 Total Memory: 12.44GiB
 Name: DESKTOP-020THD2
 ID: TMEO:GTSM:3BWM:HJJL:ZZ6L:M6PH:GFLT:HXRL:XCJI:LCXS:62RR:45HT
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No blkio throttle.read_bps_device support
WARNING: No blkio throttle.write_bps_device support
WARNING: No blkio throttle.read_iops_device support
WARNING: No blkio throttle.write_iops_device support
```

**Comment by ghost** [comment]
I couldn't figure out the best area label to add to this issue. If you have write-permissions please help me learn by adding exactly one area label.

**Comment by richlander** [comment]
Cannot repro. Try `docker build --pull` perhaps?

This is possibly a networking related error on your side.

**Comment by mcraiha** [comment]
That gives
```
"docker build" requires exactly 1 argument.
See 'docker build --help'.

Usage:  docker build [OPTIONS] PATH | URL | -

Build an image from a Dockerfile
```

and if I pull image again (with `docker pull mcr.microsoft.com/dotnet/sdk:6.0-alpine-amd64` ) it states following
```
docker pull mcr.microsoft.com/dotnet/sdk:6.0-alpine-amd64
6.0-alpine-amd64: Pulling from dotnet/sdk
Digest: sha256:ff6aa008b9cc31d60d4fef1a4424283375962793b831a53867158e09441b1127
Status: Image is up to date for mcr.microsoft.com/dotnet/sdk:6.0-alpine-amd64
mcr.microsoft.com/dotnet/sdk:6.0-alpine-amd64
```

**Comment by richlander** [comment]
Strange. Try on a different machine or network? It isn't obvious to me that the problem is with the image. I cannot repro it. Did you try `wget` within the container? That's the key test, I'd think.

I meant: `docker build --pull -t myimage .`

In any case, you clearly have the latest digest.

**Comment by mthalman** [comment]
I also cannot repro with that Dockerfile. This is very likely a host configuration issue. I see that you're using GitHub Actions in your repo. I'm guessing that if you setup an action which built this Dockerfile, it'd work. For your host machine, this is commonly related to a proxy server. For example, see https://stackoverflow.com/questions/50347517/docker-unable-to-load-the-service-index-for-source-https-api-nuget-org-v3-inde.

**Reply by mcraiha** [reply]
When I start with `docker run -it 0124e89c43c7` and then do `wget https://api.nuget.org/v3/index.json` I get following error
```
wget: bad address 'api.nuget.org'
```
but e.g. `wget https://google.com` inside same container works just fine. And wget to that nuget works just fine on host.

**Reply by richlander** [reply]
That means that there is something wrong with your environment. Something is blocking it. Can you try on a different machine? Note that no one else is reporting this issue.

**Reply by mcraiha** [reply]
Might be a nuget issue
https://github.com/NuGet/Home/issues/11879

**Reply by joelverhagen** [reply]
Hello @mcraiha, given you encountered this problem prior to when https://github.com/NuGet/Home/issues/11879 was filed, this is likely unrelated. The problem you're showing here appears more related to DNS resolution than TCP connectivity problems or bad JSON response bodies on an HTTP endpoint. 

If the issue is still occurring, could you follow the steps here?
https://docs.microsoft.com/en-us/nuget/nuget-org/nuget-org-faq#nuget.org-not-accessible

As part of that FAQ you contact NuGet.org support with diagnostic information. We'll work with you over email to investigate your specific problem.

**Reply by mcraiha** [reply]
Is there something similar for WinMTR / Fiddler part when running Linux?

**Reply by joelverhagen** [reply]
For Fiddler, there is not a great general purpose, widely adopted alternative that I am aware of. Consider using `mtr -r -c 100 api.nuget.org` for the MTR trace. Also, given this is DNS related, please provide `dig api.nuget.org` as well.

**Reply by mcraiha** [reply]
And it is working fine now.
